core/src/banking_stage/committer.rs

```diff
@@ -177,4 +189,80 @@ impl Committer {
             );
         }
     }
+
+    // If transaction's actual CU/us ratio is below cluster average COMPUTE_UNIT_TO_US_RATIO,
```

Comment:
I'm not sure this really solves our problem in a desirable way.

If the leader is creating the block with underpriced transactions, which take longer than normal for other nodes to execute during replay, then that means they were able to execute it within their 400ms.

That indicates that the leaders that lead to replay time spiking are faster than the cluster average. Unclear whether the specific underpriced transactions would be flagged as underpriced by this check, given the leader executing it is faster than average.

However, it will definitely will punish slower than average leaders. If they are already taking close to or above the cluster average for transactions which are **not** underpriced, then this adjustment makes it so they are at an even further disadvantage.

It almost feels like the underpriced check should be comparing against a validator local CU-us ratio, that tracks "what is **my** average us/CU?". If a tx is above that local average, then we can adjust since it was underpriced. I don't believe we track that information currently though.

Reply:
Agree, using _local_ execution results to compare with _cluster average_ to determine underpricing is unreliable, as mentioned in comment, that's why I added unscientific 50% taper 😭 

Another way to look at it is: 
- If leader has cluster average performance, the patch would somewhat help prevent overpacking;
- If leader is faster than average, and it still find a transaction is underpriced, then the patch would mostly help to reduce overpacking;
- If leader is slower than average, then the patch introduces the risk of lowered TPS. As Andrew pointed, slower leader therefore being punished more (which isn't necessary a bad thing), but more concern able reduced TPS for those slots.

So this patch has most positive effects when a faster leader processes many underpriced txs. And have worst effects when leader is slow. But overall effectiveness isn't clear yet.

Reply:
how are we even talking about a "cluster average time/cost ratio"?  there's no way to communicate that. the closest we can get is "local average time/cost ratio of blocks emitted by other leaders". that a slow validator "votes" for the cluster produce cheaper blocks that they can play in time amounts to communism. any operator worth their delegation would immediately revert a patch that volunteered them to produce cheaper blocks.

i'd vote to not apply this patch at all. fix the cost discrepancy instead. let the potatoes rot in the ground

Reply:
`COMPUTE_UNIT_TO_US_RATIO` is aggregated from entire cluster, but just a snapshot. But regardless, I'm also leaning to the conclusion that the patch isn't worth to be deployed, due to unjustified risk/benefit of it. 

+1 for pointing out better-than-potato validators would revert patch to get their fair capacity.

Direct mapping the the fix, which would be available 1.17?

Reply:
> Direct mapping the the fix, which would be available 1.17?

is per-cu charge in 1.16? seems like actually charging for loads is The Way :tm:. we can always reduce the cost if/when DM eventually reduces the penalty

Reply:
> seems like actually charging for loads is The Way ™️. 
we are also currently working on defining the cost of "loading account" at #31411, hope it's part of Resource Cost that will be used for Base fee. But for this case, the main cost, if i understand correctly, is the large accounts being (de)serialized many times between instructions for a transaction. Just charing a loading fee doesn't seem enough without knowing how many times the account will be copied between instructions.